### PR TITLE
Run cmake-format and fix copyright

### LINKED ISSUE #46: Clean CMake formatting

### PR DESCRIPTION
**Pull request by kdheepak**
Fixes #46

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -191,387 +248,461 @@ target_link_libraries(helics_base INTERFACE Boostlibs::core)
...
-HIDE_VARIABLE(FMT_DOC)
-HIDE_VARIABLE(FMT_INSTALL)
-HIDE_VARIABLE(FMT_PEDANTIC)
-HIDE_VARIABLE(FMT_TEST)
-HIDE_VARIABLE(FMT_USE_CPP14)
+hide_variable(FMT_DOC)
+hide_variable(FMT_INSTALL)
+hide_variable(FMT_PEDANTIC)
+hide_variable(FMT_TEST)
+hide_variable(FMT_USE_CPP14)
```
Comment (phlptp) [inline CMakeLists.txt]: Does this work,  are macros case sensitive?

File: swig/java/CMakeLists.txt [hunk 2]
```diff
@@ -11,79 +11,97 @@ include(UseJava)
...
+  if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_GREATER 3.7)
+    swig_add_library(JNIhelics TYPE MODULE LANGUAGE java SOURCES helicsJava.i)
+  else()
+    swig_add_module(JNIhelics java helicsJava.i)
+  endif()
 
-	SWIG_LINK_LIBRARIES(JNIhelics helicsSharedLib)
-	SWIG_LINK_LIBRARIES(JNIhelics ${JAVA_LIBRARIES})
+  swig_link_libraries(JNIhelics helicsSharedLib)
+  swig_link_libraries(JNIhelics ${JAVA_LIBRARIES})
```
Comment (phlptp) [inline swig/java/CMakeLists.txt]: Does capitalization matter here for these functions

**Comment by phlptp** [comment]
What is your opinion on the formatting job,  for the most part it seems to do a good job,  Some things seem a bit verbose and might be clearer on few lines, but I might be willing to live with it?

File: CMakeLists.txt (hunk 1 above)
Comment (kdheepak) [reply CMakeLists.txt]: I tested it and it built fine, no errors during the `make -j 8` process.

File: swig/java/CMakeLists.txt (hunk 2 above)
Comment (kdheepak) [reply swig/java/CMakeLists.txt]: I think in CMake `commands` are case insensitive but arguments are case sensitive.

Comment (phlptp) [reply swig/java/CMakeLists.txt]: I don't think I realized that was true for user defined functions as well as the built in ones.

**Comment by kdheepak** [comment]
I would prefer the following:

```
set(helicsShared_headers
    helics.h
    api-data.h
    ValueFederate.h
    MessageFederate.h
    MessageFilters.h
    helicsCallbacks.h)
```

to be formatted as the following instead:

```
set(
    helicsShared_headers
    helics.h
    api-data.h
    ValueFederate.h
    MessageFederate.h
    MessageFilters.h
    helicsCallbacks.h
)
```

Because if you were going to add something to it later as the last argument, the git diff would show just the right line, as opposed to showing two lines have changed, it is more noise.

There are other small things that I would like changed. 

My opinion on the matter of formatting has been that it is easier to pick a formatter that does a good enough job to make code consistent and readable, and save many person hours arguing amongst fellow developers on the right way to format code. So that's the main reason I'm in favor of having auto formatters, and not having to worry / discuss code formatting. However, if the auto formatter produces unreadable code, then that's unacceptable. I don't think that is the case here.

We can also probably make requests for the changes we want on https://github.com/cheshirekow/cmake_format, it seems to be actively developed.

**Comment by kdheepak** [comment]
They have some options that allow for setting `line-width` and `dangle-parens`. This is what I'm running the following on all files at the moment.

```
cmake-format -i {filename} --line-width 88 --dangle-parens=True
```

**Comment by phlptp** [comment]
I tend to agree with you, we can live with a bit of non-idealness for an automated solution.  And I think it is a definite improvement.  

If it does this should be a separate pr, but will this run on the .cmake files in the config directory?

**Comment by kdheepak** [comment]
I can update my script to make it run on the .cmake files in the config directory as well. At the moment they are not formatted.

**Comment by phlptp** [comment]
lets wait until this clears,  then do that in a separate PR.  It is a lot of files to go through and those might be a little more error prone.

**Pull request merged**